### PR TITLE
Delete stops

### PR DESCRIPTION
**Pull request by ForestDope**
In this PR : 

- User can delete a stop
- A delete icon is displayed on the stop item composable
- Deletion rights depends on the user role in the trip
- If the user has the rights a confirm dialog is displayed
- If the user doesnt have the rights a toast is displayed to inform him

### CODE REVIEW

File: app/src/main/java/com/github/se/wanderpals/ui/screens/trip/stops/StopItem.kt
Comment (Exino329) [inline app/src/main/java/com/github/se/wanderpals/ui/screens/trip/stops/StopItem.kt:1]: You moved `ActivityItem` composable from `DailyActivities.kt` in a new copmosable and added new things to handle deletion. I have Nothing to say about the changes you made and this file is very well documented and documented as well as for the new code. It is always nice to have readable and clear code, well done !

File: app/src/main/java/com/github/se/wanderpals/ui/screens/trip/stops/StopsList.kt [hunk 1]
```diff
@@ -54,22 +51,30 @@ import java.time.LocalDate
 @SuppressLint("UnusedMaterial3ScaffoldPaddingParameter", "WeekBasedYear")
 @OptIn(ExperimentalMaterial3Api::class)
 @Composable
-fun StopsList(stopsListViewModel: StopsListViewModel, tripId: String) {
```
Comment (Exino329) [inline app/src/main/java/com/github/se/wanderpals/ui/screens/trip/stops/StopsList.kt:57]: missing param tripId in the documentation of the method.

**Review by Exino329 — CHANGES_REQUESTED** [review]
Great code overall ! There is just few things to reconsider : 
- You can add a `CircularProgressIndicator` when stops are loading in the stop List to avoid having a blank screen
- small details concerning documentation.

File: app/src/main/java/com/github/se/wanderpals/model/viewmodel/StopItemViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+ *
+ * @param tripsRepository the repository for trips
+ * @param tripId the id of the trip
+ */
+open class StopItemViewModel(
+    private val tripsRepository: TripsRepository,
+    private val tripId: String
+) : ViewModel() {
+
+  private val _isDeleted = MutableStateFlow(false)
```
[the remark this answers is not in the document]
Comment (ForestDope) [reply app/src/main/java/com/github/se/wanderpals/model/viewmodel/StopItemViewModel.kt:22]: You're right, I will put a better name such as isDeleting

**Review by Exino329 — APPROVED** [review]
Everything looks fine !

**Pull request merged**